### PR TITLE
Bug fixes for GetCommand

### PR DESCRIPTION
**Pull request by Amadeus-Winarto**
**Changes made**
- Update DeveloperGuide for GetCommand
- Fixed unreadable and difficult to read output of GetCommand
- Fixed GetCommand behavior not matching FilterCommand
- Fixed LogicManager bug of saving StageArea when not necessary
- General code clean up
- Added integration test for VersionControl
- Added unit test for StageAreaStorage

### CODE REVIEW

File: src/main/java/seedu/academydirectory/MainApp.java [hunk 1]
```diff
@@ -63,7 +63,10 @@ public void init() throws Exception {
         initLogging(config);
 
         model = initModelManager(storage, userPrefs);
-        storage.saveStageArea(model.getStageArea());
+        if (!model.getStageArea().isEmpty()) {
+            System.out.println(model.getStageArea());
```
Comment (jianoway) [inline src/main/java/seedu/academydirectory/MainApp.java:67]: Should this be here?

File: src/test/data/VersionControlTest/VersionControlTest/Commit/24571d697e6708860610f27e36d96a12a6872d9d [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "attendance" : [ false, false, false, false, false, false, false, false, false, false, false, false ],
+    "participation" : [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
+    "assessment" : {
+      "RA2" : 4,
+      "FINAL" : 50,
+      "RA1" : 34,
+      "MIDTERM" : 13,
+      "PE" : 59
+    },
+    "tagged" : [ "colleagues", "friends" ]
```
Comment (jianoway) [inline src/test/data/VersionControlTest/VersionControlTest/Commit/24571d697e6708860610f27e36d96a12a6872d9d:16]: Is it possible to use more realistic tags in the stubs? Given it's used for test, not a priority fix.

**Review by jianoway — APPROVED** [review]
LGTM!

File: src/main/java/seedu/academydirectory/MainApp.java (hunk 1 above)
Comment (Amadeus-Winarto) [reply src/main/java/seedu/academydirectory/MainApp.java:67]: Ah fudge woops

File: src/test/data/VersionControlTest/VersionControlTest/Commit/24571d697e6708860610f27e36d96a12a6872d9d (hunk 2 above)
Comment (Amadeus-Winarto) [reply src/test/data/VersionControlTest/VersionControlTest/Commit/24571d697e6708860610f27e36d96a12a6872d9d:16]: Sure

**Pull request merged**